tests/iam/permhandler/permhandler_test.cpp

```diff
@@ -0,0 +1,211 @@
+/*
+ * Copyright (C) 2024 Renesas Electronics Corporation.
+ * Copyright (C) 2024 EPAM Systems, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "aos/common/tools/buffer.hpp"
+#include "mocks/subjectsobservermock.hpp"
+#include <gtest/gtest.h>
+
+#include "aos/iam/permhandler.hpp"
+#include <iostream>
+
+using namespace aos;
+using namespace aos::iam::permhandler;
+using namespace testing;
+
+/***********************************************************************************************************************
+ * Suite
+ **********************************************************************************************************************/
+
+class PermHandlerTest : public Test {
+protected:
+    void SetUp() override
+    {
+        Log::SetCallback([](LogModule module, LogLevel level, const String& message) {
+            static std::mutex sLogMutex;
+
+            std::lock_guard<std::mutex> lock(sLogMutex);
+
+            std::cout << level.ToString().CStr() << " | " << module.ToString().CStr() << " | " << message.CStr()
+                      << std::endl;
+        });
+    }
+
+    void TearDown() override { }
+
+    PermHandler mPermHandler;
+};
+
+/***********************************************************************************************************************
+ * Consts
+ **********************************************************************************************************************/
+
+/***********************************************************************************************************************
+ * Tests
+ **********************************************************************************************************************/
+
+TEST_F(PermHandlerTest, RegisterInstanceSucceeds)
+{
+    InstanceIdent                                instanceIdent;
```

Comment:
var definition should be separated with other operators. This function should look like:
```c++
TEST_F(PermHandlerTest, RegisterInstanceSucceeds)
{
    InstanceIdent                                instanceIdent;
    StaticArray<FunctionalServicePermissions, 1> perms;

    auto err = mPermHandler.RegisterInstance(instanceIdent, perms);
    EXPECT_TRUE(err.mError.IsNone()) << err.mError.Message();
    ASSERT_FALSE(err.mValue.IsEmpty());
}
```